src/CHIPBackend.hh

```diff
@@ -536,7 +536,7 @@ class CHIPQueue;
 class CHIPContext;
 class CHIPDevice;
 
-class CHIPEvent {
```

Comment:
Why didn't I come up with that.. 

Anyways, I think this is an issue because of our modified HIP headers. It didn't occur to me make CHIP classes inherit from these internal structures. iirc, `ihipEvent_t` is defined in the original HIP headers. 

Having said that, structures that start with `i` are internal and should not be used by external libraries

Reply:
> Having said that, structures that start with i are internal and should not be used by external libraries

I thought that too initially, but after seeing the PyTorch code and [this](https://docs.amd.com/bundle/HIP_Supported_CUDA_API_Guide_v5.3/page/CUDA_Runtime_API_functions_supported_by_HIP.html) (which lists the `ihipEvent_t` as supported) it seems more that the clients may use the `ihipEvent_t` as a forward declaration.